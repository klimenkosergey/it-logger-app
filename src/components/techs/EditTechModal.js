import React from 'react';
import M from 'materialize-css/dist/js/materialize';

import TechForm from './TechForm';

const EditTechModal = () => {
  const onSubmit = (firstName, lastName) => {
    const modalElem = document.querySelector('#edit-tech-modal');
    const instance = M.Modal.getInstance(modalElem);

    if (firstName === '' || lastName === '') {
      M.toast({ html: 'Please enter first and last names' });
    } else {
      instance.close();
    }
  };

  return (
    <div id='edit-tech-modal' className='modal'>
      <TechForm onSubmit={onSubmit} />
    </div>
  );
};

export default EditTechModal;
