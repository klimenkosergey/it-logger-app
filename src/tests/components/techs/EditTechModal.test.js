import React from 'react';
import { shallow } from 'enzyme';

import EditTechModal from '../../../components/techs/EditTechModal';

it('should render EditTechModal correctly', () => {
  const wrapper = shallow(<EditTechModal />);
  expect(wrapper).toMatchSnapshot();
});
