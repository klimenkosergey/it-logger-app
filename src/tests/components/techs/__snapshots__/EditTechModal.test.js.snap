// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditTechModal correctly 1`] = `
<div
  className="modal"
  id="edit-tech-modal"
>
  <TechForm
    onSubmit={[Function]}
  />
</div>
`;
